pennylane_lightning/core/src/simulators/base/StateVectorBase.hpp

```diff
@@ -44,6 +44,14 @@ template <class PrecisionT, class Derived> class StateVectorBase {
   private:
     size_t num_qubits_{0};
 
+  protected:
+    /**
+     * @brief Set the number of qubits represented by the statevector data.
+     *
+     * @return std::size_t
+     */
+    void setNumQubits(size_t qubits) noexcept { num_qubits_ = qubits; }
```

Comment:
@AmintorDusko suggestion to make `num_qubits_` protected instead of `private` works fine if you think this's a better approach than adding `setNumQubits` with the default throwing exception. I like the latter slightly  more than making a base class value protected.

Reply:
I leave the final decision to you guys (@AmintorDusko @mlxd) as I don't have any strong opinion on this. :100:

Reply:
Since this is still hidden under the hood, I prefer not to add methods that aren't needed except for a single location's use. Fine with the move to protected here if it solves the issue.

We can always revisit a more wide-spread integration of the dynamic functionality later, but let's use this approach for now.

Reply:
Sounds good! Yes, the protected `num_qubits_` works too.